demo_nodes_py/demo_nodes_py/topics/listener_qos.py

```diff
@@ -25,11 +26,15 @@ class ListenerQos(rclpy.Node):
 
     def __init__(self, qos_profile):
         super().__init__('listener_qos')
+        if qos_profile.reliability is QoSReliabilityPolicy.RMW_QOS_POLICY_RELIABILITY_RELIABLE:
+            self.get_logger().info('Reliable listener')
```

Comment:
It's not obvious to me that this is actually performing a logging action compared to the API suggested originally in https://github.com/ros2/rclpy/issues/103, maybe just calling the function `loginfo` or a `log` function taking a severity keyword would be clearer.
I know the recent discussion was focused on the c++ API but how do we expect the API to look like for the users in python?

Reply:
the `.log(,INFO)` option is also available on this logger, following https://github.com/ros2/rclpy/issues/103#issuecomment-324148790 (which came from a team discussion), `.info()` is offered for convenience. Both options mirror what is offered by python logging, we seem to prefer the `.info()`: https://github.com/ros2/ros2cli/blob/9fce6a08381bb67cf17e7160b0b2c6556ffec1e6/ros2cli/ros2cli/entry_points.py#L87

Reply:
Ok so the goal is to use an API as close as python native logging as possible rather than something closer to ROS 1. We do provide both but will use this one in our examples. Fair enough.
With https://github.com/ros2/rclpy/pull/148#issuecomment-344781872 it now make sense why we don't something like `self.mylogger.loginfo()`

Reply:
Following up on this, if I understand correctly your initial concern was referring to how we would use the generic logging functions (e.g. `rclpy.logging.loginfo()`) as opposed to log calls being made on specific logger objects (which weren't around in ROS 1).

On the use of `mylogger.info()` as opposed to `mylogger.log(, severity=INFO)`, as you noted I've chosen to mirror the typical usage of native python logging. https://docs.python.org/3/howto/logging-cookbook.html, for example, uses `mylogger.info()` in all cases except where the severity is chosen programatically, for which case the `mylogger.log(, severity=INFO)` form is available